### DISCUSSION TITLE
www files overwritten after reboot/ newly started

### DISCUSSION THREAD

**Discussion by BigBobDalton** [gtxaspec/wz_mini_hacks#647] [Q&A] [answered]
When I reboot my V3 camera the index.html file in the directory opt/wz_mini/www defaults to web interface version with no menu.  I have to nano edit and update it with the following after each reboot or power cycle.

Is there a way to protect this file or could someone direct me to where the default file is that keeps overwriting these changes?

<html>
  <head><title>WZ Mini Web Tools</title></head>
  <body>
  <h1>wz_mini web server</h1>
  <a href="https://github.com/gtxaspec/wz_mini_hacks">wz_mini GitHub</a>

    <p><a href="cgi-bin/config.cgi">Wz Mini Configuration Editor</a></p>
    <p><a href="cgi-bin/cam.cgi">Camera Configuration</a></p>
    <p><a href="car/car.html">Car Interface</a></p>
    <p><a href="cgi-bin/jpeg.cgi">Current Screen</a></p>
    <p><a href="multicam.html">Multi-Cam Viewer</a></p>
    <p><a href="cgi-bin/diagnostics.cgi">Diagnostics</a></p>
  </body>
</html>

Thanks

**Comment by BigBobDalton** [comment]
This is the index.html upon reboot

<html>
<head>
<title>WZ Mini Web Tools</title>
<style type="text/css">
.server_LINK {
 display:block;
 
}
#display_BAR {
    position: fixed;
    bottom: 0;
    background-color: blue;
    border-top: 1px solid lightblue;
    color: #fff;
    display: flex;
    width: 100%;
    height: var(--display_bar_height);
}
#display_BAR A , #display_BAR A:visited {
 color:white;
}
</style>
</head>
<body>
<h1>Wyze Mini Web Sever Menu</h1>
<div id="display_BAR"><div class="github_link"><a target="_new" href="https://github.com/gtxaspec/wz_mini_hacks">GitHub: wz_mini_hack</a></div></div>
</body></html>

**Comment by virmaior** [comment, marked as answer]
The overwriting is intentional. Index.html is getting composed on reboot.

There's an issue I've put in a pull request to resolve ( https://github.com/gtxaspec/wz_mini_hacks/pull/646)

1. In the current version, it's not displaying a message it should telling you to set WEB_SERVER_OPTIONS (didn't pipe this correctly)
2. The WEB_SERVER_OPTIONS in wz_mini.conf populates the list of pages there (if someone disables a page in  WEB_SERVER_OPTIONS, it becomes inaccessible and isn't listed).

If you're doing something else with the page edit `index.top.html` manually to include other things you're doing.

my apologies for any confusion or difficulty this is causing

**Reply by vic4news** [reply]
Another issue related to this is that the multicam link from the web menu doesn’t work. Also “Server” is misspelled in the title. 

Wyze Mini Web Sever Menu

[Wyze Camera Core Config](http://192.168.0.90/cgi-bin/cam.cgi)
[Wz Mini Configuration](http://192.168.0.90/cgi-bin/config.cgi)
[Car Tool](http://192.168.0.90/car.html)
[Check Recording](http://192.168.0.90/cgi-bin/status.cgi?test=recording)
[Multi-Cam](http://192.168.0.90/mulitcam.html)
[Go2RTC Server](http://192.168.0.90/#Go2RTC)

if I select the multicam link I get the error 404 Not Found, although the file opt/wz_mini/www/multicam.html does exist

**Reply by virmaior** [reply]
Got the server one!

In the latest version

multicam is put in its own directory  as `/multicam/`
```
multicam) echo '<a class="server_LINK" href="/multicam/" >Multi-Cam Viewer</a>' >> $target       ;;
```

is that broken?

**Reply by vic4news** [reply]
This is what I have in the www directory 

[root@BACKYARD:~]# cd opt/wz_mini/www/
[root@BACKYARD:www]# ls -l
total 288
-rwxrwxrwx    1 root     root          1549 Sep 30 18:46 cam.js
drwxrwxrwx    2 root     root         32768 Oct  3 12:46 car
drwxrwxrwx    2 root     root         32768 Oct  3 12:46 cgi-bin
-rwxrwxrwx    1 root     root          3255 Sep 30 18:46 config.js
-rwxrwxrwx    1 root     root           429 Sep 30 18:46 feed.js
-rwxrwxrwx    1 root     root          1049 May 13  2022 index.html
-rwxrwxrwx    1 root     root           432 Oct  5 13:13 index.top.html
drwxrwxrwx    2 root     root         32768 Oct  3 12:46 multicam
-rwxrwxrwx    1 root     root          9621 Sep 30 18:46 multicam.html

[root@BACKYARD:www]# ls multicam
index.css   index.html  index.js
[root@BACKYARD:www]#

**Reply by virmaior** [reply]
the link is misspelled as "mulitcam.html" but the more correct link is to /mulitcam/ which is a more recent version.

**Reply by vic4news** [reply]
Thanks. Correcting the spelling in the link resolves the problem
